### ISSUE TITLE
PSyclone is now unfriendly to case-insensitive filesystems

### ISSUE THREAD

**Issue by nmnobre** [stfc/PSyclone#2239]
Hi @hiker,

f631964 creates files whose names only differ from previously existing ones by their upper/lower-case extensions.
This is a problem in case-insensitive filesystems, can we consider some renaming?

Cheers,
Nuno

**Comment by hiker** [comment]
Dang, that's a bit of a problem. We need to pre-process the files before using PSyclone on them (since PSyclone does not support preprocessing files), but we also want to store the original files (so we can simply change e.g. the size of certain variables by re-preprocessing these files).
But yes, case-insensitive FS will have a problem :(
I see three solutions:
1. We use a different extension ... though that's really ugly (e.g. F90 and F95 for the preprocessed files) - that's so ugly, that my fingers barely want to type this :)
2. We officially do not support case-insensitive FS :)  OK, only joking 
3. We put the pre-processed files in a different subdirectory. And that directly cannot be under `.../infrastructure` (since the module manager searches recursively, it would then find two files implementing the same module, which can only lead to chaos). Or maybe `.../infrastructure/preprocessed` and `.../infrastructure/original`?

Does anyone have a better idea?

**Comment by hiker** [comment]
> > 1. We use a different extension ... though that's really ugly (e.g. F90 and F95 for the preprocessed files) - that's so ugly, that my fingers barely want to type this :)
> 
> Is it LFRic that uses `.x90`? Could that be a possibility, or does it still make you a bit uneasy?
That's different :) .x90 files indicate that the file contains invokes and need to be processed by PSyclone before compilation.

As part of the kernel extraction/driver creation feature, PSyclone now needs to be able to parse files that are usually not processed by PSyclone: Eg. a PSyclone kernel might call a function from an infrastructure module, which uses a value stored in a module variable ... then the kernel extraction needs to store this value, and in order to do this, it needs to parse these files, which are usually not touched by PSyclone. And PSyclone can only parse pre-processed files. And in order for coverage and examples to work, we need to have the pre-processed files somewhere.

> 
> > 2. We officially do not support case-insensitive FS :)  OK, only joking
> 
> I understand most people work under GNU/Linux so that's unlikely to upset many people, but for those coding under Windows or macOS... (like me :P)

Well, I am on Windows (WSL2 :)  but used a mounted windows FS on a ubuntu VM in the past, and had the exactly same problem), but yes, I do remember the OSX problems (isn't there an option to make the file system case aware??) Anyway, I really was only joking here .

**Comment by nmnobre** [comment]
> Well, I am on Windows (WSL2 :) but used a mounted windows FS on a ubuntu VM in the past, and had the exactly same problem), but yes, I do remember the OSX problems (isn't there an option to make the file system case aware??) Anyway, I really was only joking here .

There is, even though it's not default. I know, no worries. :)

**Comment by hiker** [comment]
@nmnobre , is this urgent (i.e. does it prevent you from working)? I would prefer to discuss with the team first before touching/moving the infrastructure files.

**Comment by nmnobre** [comment]
Yeah, no, it's not really urgent. I guess it's been reported now, so people shouldn't be surprised next time they clone the repo on a case-insensitive filesystem that the working directory will be instantly and unavoidably dirty.

**Comment by hiker** [comment]
I am wondering if it might be worth moving the infrastructure files under ``external``. While obviously this is atm not really an external repo, my understanding is that UKMO is going this way, so if we move (some) files around, maybe now is a good time to prepare for this?

**Comment by arporter** [comment]
> I am wondering if it might be worth moving the infrastructure files under `external`. While obviously this is atm not really an external repo, my understanding is that UKMO is going this way, so if we move (some) files around, maybe now is a good time to prepare for this?

I like that idea.
